src/pudl/output/eia930.py

```diff
@@ -26,30 +52,44 @@
 )
 def _out_eia930__hourly_subregion_demand(
     core_eia930__hourly_subregion_demand: pd.DataFrame,
+    core_eia__codes_balancing_authorities: pd.DataFrame,
 ) -> pd.DataFrame:
     """Adds timezone column and combined ID with BA/subregion used for imputation."""
-    core_eia930__hourly_subregion_demand["timezone"] = (
-        core_eia930__hourly_subregion_demand["balancing_authority_code_eia"].map(
-            BA_TIMEZONE_MAP
-        )
+    core_eia930__hourly_subregion_demand = _add_timezone(
+        core_eia930__hourly_subregion_demand,
+        core_eia__codes_balancing_authorities,
     )
+
     core_eia930__hourly_subregion_demand["combined_subregion_ba_id"] = (
         core_eia930__hourly_subregion_demand["balancing_authority_code_eia"]
         + core_eia930__hourly_subregion_demand["balancing_authority_subregion_code_eia"]
     )
     return core_eia930__hourly_subregion_demand
 
 
-imputed_subregion_demand_assets = impute_timeseries_asset_factory(  # pragma: no cover
-    input_asset_name="_out_eia930__hourly_subregion_demand",
-    output_asset_name="out_eia930__hourly_subregion_demand",
-    years_from_context=lambda context: [
+def _years_from_context(context) -> list[int]:
+    return [
         int(half_year[:4])
         for half_year in context.resources.dataset_settings.eia.eia930.half_years
-        if "2025" not in half_year
-    ],
+    ]
+
+
+imputed_subregion_demand_assets = impute_timeseries_asset_factory(
+    input_asset_name="_out_eia930__hourly_subregion_demand",
+    output_asset_name="out_eia930__hourly_subregion_demand",
+    years_from_context=_years_from_context,
     value_col="demand_reported_mwh",
     imputed_value_col="demand_imputed_pudl_mwh",
     id_col="combined_subregion_ba_id",
-    settings=ImputeTimeseriesSettings(method_overrides={2019: "tnn"}),
+    settings=ImputeTimeseriesSettings(method_overrides={2019: "tnn", 2025: "tnn"}),
```

Comment:
Now that I implemented the padding before imputation, we can include 2025 data, but 2025 was failing in the same way as 2019.